CMakeLists.txt

```diff
@@ -10,35 +10,69 @@
 project(xcbuild C CXX)
 
 cmake_minimum_required(VERSION 3.0)
-set(CMAKE_MACOSX_RPATH 1)
+cmake_policy(SET CMP0054 NEW)
 
-if ("${CMAKE_SYSTEM_NAME}" MATCHES "Darwin")
+set(BUILD_SHARED_LIBS True CACHE BOOL "Build shared libraries." FORCE)
+
+# Platform options.
+if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
+  # Export all symbols to avoid needing a `.def` file or `__dllexport`.
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS True)
+
+  if ("${CMAKE_CXX_PLATFORM_ID}" STREQUAL "MinGW")
+    # Use MinGW formatting rather than MSVCRT for C99 support.
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D__USE_MINGW_ANSI_STDIO=1")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__USE_MINGW_ANSI_STDIO=1")
+  endif ()
+elseif ("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+  set(CMAKE_MACOSX_RPATH True)
   set(CMAKE_INSTALL_RPATH "@executable_path/../lib")
-else ()
+elseif ("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
   set(CMAKE_INSTALL_RPATH "$ORIGIN/../lib")
 endif ()
 
 # Output into root build directory.
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+# Language version.
```

Comment:
I'm not 100% sure. It might be a per-project thing. I don't think I actually know how CMake variables work exactly :p